### PR TITLE
Adding body landmarks

### PR DESCRIPTION
**Pull request by Rauf-Kurbanov**

### CODE REVIEW

File: face_api_dataset/dataset.py [hunk 1]
```diff
@@ -58,21 +62,53 @@ class Modality(Enum):
...
+
+    **Type**: `List[Type[str, int, int]`.
+    """
+    LANDMARKS_COCO = auto()
+    """
+    COCO Whole Body landmarks. Each landmark is given by name and two coordinates (name, x,y) in pixels.
+
+    **Type**: `List[Type[str, int, int]`.
+    """
+    LANDMARKS_MPEG4 = auto()
```
Comment (adavydow) [inline face_api_dataset/dataset.py:97]: Let's add 3D coordinates for body landmarks as well.

File: face_api_dataset/dataset.py [hunk 2]
```diff
@@ -58,21 +62,53 @@ class Modality(Enum):
     
     **Type**: `ndarray[uint16]`. **Channels**: 1.
     """
-    LANDMARKS = auto()
+    LANDMARKS_IBUG68 = auto()
+    """
+    iBUG-68 landmarks. Each landmark is given by name and two coordinates (name, x,y) in pixels.
+    Each keypoint is a 2D projection of a 3D landmark. 
+    
+    **Type**: `List[Type[str, int, int]`.
```
Comment (adavydow) [inline face_api_dataset/dataset.py:70]: Type=Tuple?

File: face_api_dataset/dataset.py [hunk 3]
```diff
@@ -58,21 +62,53 @@ class Modality(Enum):
...
+
+    **Type**: `List[Type[str, int, int]`.
+    """
+    LANDMARKS_MPEG4 = auto()
+    """
+    MPEG4 landmarks. Each landmark is given by name and two coordinates (name, x,y) in pixels.
+
+    **Type**: `List[Type[str, int, int]`.
+    """
+    LANDMARKS_3D_IBUG68 = auto()
```
Comment (adavydow) [inline face_api_dataset/dataset.py:103]: Let's update 3D landmarks to tuple format as well.

File: face_api_dataset/dataset.py [hunk 4]
```diff
@@ -58,21 +62,53 @@ class Modality(Enum):
...
     iBUG-68 landmarks in 3D. Each landmark is given by three coordinates (x,y,z) in camera space.
 
     **Type**: `ndarray[float64]`. **Dimensions**: (68, 3).
     """
     PUPILS = auto()
     """
-    Coordinates of pupils. Each pupil is given by two coordinates (x,y) in pixels.
+    Coordinates of pupils. Each pupil is given by name and two coordinates (name, x,y) in pixels.
     
     **Type**: `ndarray[float64]`. **Dimensions**: (2, 2).
```
Comment (adavydow) [inline face_api_dataset/dataset.py:113]: Type update needed.

File: face_api_dataset/dataset.py [hunk 5]
```diff
@@ -58,21 +62,53 @@ class Modality(Enum):
...
+    **Type**: `List[Type[str, int, int]`.
+    """
+    LANDMARKS_CONTOUR_IBUG68 = auto()
     """
-    iBUG-68 landmarks. Each landmark is given by two coordinates (x,y) in pixels.
+    iBUG-68 contour landmarks. Each landmark is given by two coordinates (name, x,y) in pixels.
+    Each keypoint is defined in a similar manner to human labelers marking 2D face kepoints.
     
-    **Type**: `ndarray[float64]`. **Dimensions**: (68, 2).
+    **Type**: `List[Type[str, int, int]`. Should have 68 points.
```
Comment (adavydow) [inline face_api_dataset/dataset.py:77]: Should have no more than 68 points (only visible points are included).

File: face_api_dataset/dataset.py (hunk 2 above)
Comment (adavydow) [reply face_api_dataset/dataset.py:70]: I believe all coordinates are float, not int?

File: face_api_dataset/dataset.py (hunk 4 above)
Comment (adavydow) [reply face_api_dataset/dataset.py:113]: Let's update PUPILS_3D as well.

**Review by adavydow — CHANGES_REQUESTED** [review]
Some comments added.

**Comment by adavydow** [comment]
TODO:
1. Update Dataset (use only heads with bodies, both head and full body shots).
2. Regenerate documentation.

File: face_api_dataset/dataset.py (hunk 1 above)
Comment (Rauf-Kurbanov) [reply face_api_dataset/dataset.py:97]: done

File: face_api_dataset/dataset.py (hunk 3 above)
Comment (Rauf-Kurbanov) [reply face_api_dataset/dataset.py:103]: done

File: face_api_dataset/dataset.py (hunk 4 above)
Comment (Rauf-Kurbanov) [reply face_api_dataset/dataset.py:113]: done

File: face_api_dataset/dataset.py (hunk 5 above)
Comment (Rauf-Kurbanov) [reply face_api_dataset/dataset.py:77]: fixed

**Comment by Rauf-Kurbanov** [comment]
@adavydow not sure if we want to keep the actual dataset in git, or just provide the API spec. Added both just in case.

**Review by adavydow — COMMENTED** [review]
LGTM. TODO: regenerate docs.

**Comment by adavydow** [comment]
> @adavydow not sure if we want to keep the actual dataset in git, or just provide the API spec. Added both just in case.

I'd like to keep it (if anybody without access to our API will want to have a look). Probably it's reasonable to move to git-lfs

**Pull request closed without merging**